js/hotgraphicPopupView.js

```diff
@@ -0,0 +1,196 @@
+
+            if (!hasAccessibility) {
+                $(window).on("keyup", this.onKeyUp);
+            }
+        },
+
+        removeEscapeKey: function() {
+            $(window).off("keyup", this.onKeyUp);
+        },
+
+        onKeyUp: function(event) {
+            if (event.which != 27) return;
+            event.preventDefault();
+            this.closePopup();
+        },
+
+        previousHotGraphic: function () {
+            var currentIndex = this.model.getActiveItem().get('_index');
+            var canCycleThroughPagination = this.model.get('_canCycleThroughPagination');
+            var itemLength = this.model.get('_items').length;
+
+            if (currentIndex === 0 && !canCycleThroughPagination) {
+                return;
+            } else if (currentIndex === 0 && canCycleThroughPagination) {
+                currentIndex = itemLength;
+            }
+
+            this.setItemState(currentIndex-1);
+
+            this.$('.hotgraphic-item.active').hide().removeClass('active');
+            this.$('.hotgraphic-item').eq(currentIndex-1).show().addClass('active');
+            this.$currentLabel.html(currentIndex);
+            this.$('.hotgraphic-popup-inner').a11y_on(false);
+
+            this.applyNavigationClasses(currentIndex-1);
+            this.handleFocus(true);
+            this.updatePageCount();
+        },
+
+        nextHotGraphic: function () {
+            var currentIndex = this.model.getActiveItem().get('_index');
+            var canCycleThroughPagination = this.model.get('_canCycleThroughPagination');
+            var itemLength = this.model.get('_items').length;
+
+            if (currentIndex === (itemLength-1) && !canCycleThroughPagination) {
+                return;
+            } else if (currentIndex === (itemLength-1) && canCycleThroughPagination) {
+                currentIndex = -1;
+            }
+
+            this.setItemState(currentIndex+1);
+
+            this.$('.hotgraphic-item.active').hide().removeClass('active');
+            this.$('.hotgraphic-item').eq(currentIndex+1).show().addClass('active');
+            this.$currentLabel.html(currentIndex+2);
+            this.$('.hotgraphic-popup-inner').a11y_on(false);
+
+            this.applyNavigationClasses(currentIndex+1);
+            this.handleFocus(true);
+            this.updatePageCount();
+        },
+
+        setItemState: function(index) {
+            this.model.getActiveItem().set('_isActive', false);
+            var nextItem = this.model.getItem(index);
+            nextItem.set({
+                '_isActive': true,
+                '_isVisited': true
+            });
+        },
+
+        updatePageCount: function() {
+            var globals = Adapt.course.get("_globals");
+            var custom = (globals._components._hotgraphics && globals._component._hotgraphic.popupPagination);
+            var template = custom || '{{itemNumber}} / {{totalItems}}';
```

Comment:
Do you need to have a default here now that the grunt task will automatically populate course.json with the globals in properties.schema?

Reply:
for some reason I thought it was a fairly recent addition but it [looks like it's been in Adapt since Oct 2015](https://github.com/adaptlearning/adapt_framework/commits/master/grunt/tasks/schema-defaults.js)!!